src/main/java/core/basesyntax/Shape.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax;
+
+public abstract class Shape {
+
+    private String name;
+    private String color;
+    private StringBuilder sb = new StringBuilder();
+
+    public Shape(String name) {
+        this.name = name;
+        this.color = randomColor();
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public abstract double calculateArea();
+
+    public abstract double perimeter();
+
+    public abstract void draw();
+
+    public String info() {
+        return sb.append("Shape: ").append(this.getName())
+        .append(", color: " + this.getColor()).toString();
+    }
+
+    private String randomColor() {
```

Comment:
move this logic to your FIgureProducer or create separate class

Reply:
done!